### PR TITLE
[amazonechocontrol] fix unnecessary refresh and add text commands

### PR DESCRIPTION
**Pull request by J-N-K**
This fixes an issue with lagging announcements due to traffic caused by unnecessary refreshes and also adds a new feature: text commands. With text commands openhab can execute commands on alexas as if they were spoken by the user. It would be really helpful to have this in OH3.

### CODE REVIEW

File: bundles/org.openhab.binding.amazonechocontrol/src/main/resources/OH-INF/i18n/amazonechocontrol_de.properties [hunk 1]
```diff
@@ -88,6 +88,9 @@ channel-type.amazonechocontrol.textToSpeech.description = Spricht den Text (Nur
 channel-type.amazonechocontrol.textToSpeechVolume.label = Sprich Lautst�rke
 channel-type.amazonechocontrol.textToSpeechVolume.description = Lautst�rke des Sprich Kanals. Wenn 0 wird die aktuelle Lautst�rke verwendet.
 
+channel-type.amazonechocontrol.textCommand.label = Befehl
+channel-type.amazonechocontrol.textCommand.description = F�hrt einen Befehl aus (Nur schreiben). Der Befehl kann wie ein gesprochener Befehl ausgef�hrt werden.
```
Comment (mgeramb) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/resources/OH-INF/i18n/amazonechocontrol_de.properties:92]: Vorschlag: "Führt einen Befehl aus (Nur schreiben). Der Text wird wie ein gesprochener Befehl ausgeführt"

**Comment by jaywiseman1971** [comment]
@J-N-K,

I've noticed a lot changes over the last few weeks to the milestone 3.0 branch with the Amazon binding.  Is there a plan to put these bug fix changes into the milestone 2.5.x branch?

Best, Jay

**Comment by J-N-K** [comment]
No, at least not from my side. This can‘t be backported anyway, no new features for 2.5.x

**Comment by jaywiseman1971** [comment]
@J-N-K 

Here's one of those bugs I was wondering about with 2.5.x branch.

https://github.com/openhab/openhab-addons/pull/9057#event-4039712220

Best, Jay

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java [hunk 2]
```diff
@@ -1441,6 +1446,58 @@ private void sendTextToSpeech() {
...
+                        Map<String, Object> parameters = new HashMap<>();
+                        parameters.put("text", text);
+                        executeSequenceCommandWithVolume(devices, "Alexa.TextCommand", parameters,
+                                textCommand.ttsVolumes, textCommand.standardVolumes);
+                    }
+                } catch (Exception e) {
+                    logger.warn("send textCommand fails with unexpected error", e);
+                }
+            }
+            iterator.remove();
```
Comment (cpmeister) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java:1493]: You are only removing the last command?

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java [hunk 3]
```diff
@@ -1441,6 +1446,58 @@ private void sendTextToSpeech() {
...
+            Iterator<TextCommand> iterator = textCommands.values().iterator();
+            while (iterator.hasNext()) {
+                TextCommand textCommand = iterator.next();
+                try {
+                    List<Device> devices = textCommand.devices;
+                    if (!devices.isEmpty()) {
+                        String text = textCommand.text;
+                        Map<String, Object> parameters = new HashMap<>();
+                        parameters.put("text", text);
+                        executeSequenceCommandWithVolume(devices, "Alexa.TextCommand", parameters,
```
Comment (cpmeister) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java:1486]: Would this be a good place to use `Map.of`? Or do you need the map to be a HashMap instance?

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/EchoHandler.java [hunk 4]
```diff
@@ -1083,6 +1103,10 @@ public void updateState(AccountHandler accountHandler, @Nullable Device device,
                 updateTextToSpeech = false;
                 updateState(CHANNEL_TEXT_TO_SPEECH, new StringType(""));
             }
+            if (updateTextCommand) {
+                updateTextCommand = false;
+                updateState(CHANNEL_TEXT_COMMAND, new StringType(""));
```
Comment (cpmeister) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/EchoHandler.java:1108]: Probably should make the same changes elsewhere as well.
```suggestion
                updateState(CHANNEL_TEXT_COMMAND, StringType.EMPTY);
```

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java (hunk 2 above)
Comment (J-N-K) [reply bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java:1493]: Oops. Remove wrong `}`...

**Review by cpmeister — APPROVED** [review]

**Comment by cpmeister** [comment]
Can you fix the sign-off?

**Comment by J-N-K** [comment]
Unfortunately not. That commit is totally authored by Tom and therefore signed-off from him.

**Pull request merged**